### PR TITLE
New implementation of Word2Vec

### PR DESCRIPTION
**Pull request by maximilienroberti**
New implementation of Word2Vec.
For more information please consult the documentation

### CODE REVIEW

File: documentation/CONFIG_PARAMETERS.md [hunk 1]
```diff
@@ -302,9 +298,116 @@ TODO
...
+    Used mainly to evaluate the performance of word2vec. 
+    The probabilistic model will compute the true probability that a context word to appear given a certain center word.
+    <code>P(context_word|center_word) = (num. of time the pair context_word-center_word appears)/(num. of time center_word appears)</code>.
+    Set to <code>0</code> by default.
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-0pky"><code>output_prob</code></td>
+    <td class="tg-0pky"><code>0</code>, <code>1</code></td>
+    <td class="tg-0pky">If set to <codde>1</codde>, the models output the probability hat a context word appears,
```
Comment (daanraman) [inline documentation/CONFIG_PARAMETERS.md:334]: code instead of codde

File: documentation/CONFIG_PARAMETERS.md [hunk 2]
```diff
@@ -302,9 +298,116 @@ TODO
...
+    <td class="tg-0pky"><code>size_window</code></td>
+    <td class="tg-0pky"><code>Int</code></td>
+    <td class="tg-0pky">Size of the context window.
+    Note that as you increase the size window, the number of center word - context word combination will increase. 
+    It will then result in a augmentation of memory size and computation time.</td>
+  </tr>
+  <tr>
+    <td class="tg-0pky"><code>min_uniq_word_occurrence</code></td>
+    <td class="tg-0pky"><code>Int</code></td>
+    <td class="tg-0pky">If a word appears less than <code>min_uniq_word_occurrence</code> value, it will be replaced by
```
Comment (daanraman) [inline documentation/CONFIG_PARAMETERS.md:358]: value -> times

File: documentation/CONFIG_PARAMETERS.md [hunk 3]
```diff
@@ -302,9 +298,116 @@ TODO
...
+    <td class="tg-0pky">Size of the context window.
+    Note that as you increase the size window, the number of center word - context word combination will increase. 
+    It will then result in a augmentation of memory size and computation time.</td>
+  </tr>
+  <tr>
+    <td class="tg-0pky"><code>min_uniq_word_occurrence</code></td>
+    <td class="tg-0pky"><code>Int</code></td>
+    <td class="tg-0pky">If a word appears less than <code>min_uniq_word_occurrence</code> value, it will be replaced by 
+    the 'UNKNOWN' word. Set to <code>1</code> by default.
+    Note that as it reduces the vocabulary size of the model, it reduce the memory size.</td>
```
Comment (daanraman) [inline documentation/CONFIG_PARAMETERS.md:360]: it reduces*

File: documentation/CONFIG_PARAMETERS.md [hunk 4]
```diff
@@ -478,9 +581,9 @@ To have more information about the configuration of one analyzer, visit the page
   </tr>
 </table>
 
-### Usual model paramaters
+### Usual model parameters
```
[the remark this answers is not in the document]
Comment (maximilienroberti) [reply documentation/CONFIG_PARAMETERS.md:584]: Usual parameters a title already there and is referred to the section "[Usual parameters](https://github.com/NVISO-BE/ee-outliers/blob/master/documentation/CONFIG_OUTLIERS.md#usual-model-parameters)" in "[Building detection use cases](https://github.com/NVISO-BE/ee-outliers/blob/master/documentation/CONFIG_OUTLIERS.md#usual-model-parameters)" tutorial. Can be changed to "Common model parameters". What do you think?

[the remark this answers is not in the document]
Comment (maximilienroberti) [reply documentation/CONFIG_PARAMETERS.md:584]: Those parameters are common to "terms", "metrics" and "word2vec".

[the remark this answers is not in the document]
Comment (daanraman) [reply documentation/CONFIG_PARAMETERS.md]: I propose to just delete that "usual model parameters" section, doesn't really mean anything there when I see it.

**Pull request merged**